File: src/main/java/com/mudhut/nudge/users/services/UserService.java
package com.mudhut.nudge.users.services;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import com.mudhut.nudge.users.entities.User;
import com.mudhut.nudge.users.repositories.UserRepository;
import jakarta.persistence.EntityNotFoundException;

import java.util.List;

@Service
public class UserService {

    @Autowired
    private UserRepository userRepository;

    public User getUserById(Long id) {
        return userRepository.findById(id)
                .orElseThrow(() -> new EntityNotFoundException("User not found with id: " + id));
    }

    public List<User> getAllUsers() {
        return userRepository.findAll();
    }

    public User updateUser(Long id, User userDetails) {
        User user = getUserById(id);
        return userRepository.save(user);
    }

    public void deleteUser(Long id) {
        User user = getUserById(id);
        userRepository.delete(user);
    }

}

File: src/main/kotlin/com/mudhut/nudge/users/services/UserService.kt
package com.mudhut.nudge.users.services

import com.mudhut.nudge.users.entities.User
import com.mudhut.nudge.users.repositories.UserRepository
import jakarta.persistence.EntityNotFoundException
import org.springframework.stereotype.Service

@Service
class UserService(
    private val userRepository: UserRepository
) {

    fun getUserById(id: Long): User =
        userRepository.findById(id)
            .orElseThrow { EntityNotFoundException("User not found with id: $id") }

    fun getAllUsers(): List<User> = userRepository.findAll()

    fun updateUser(id: Long, userDetails: User): User {
        val user = getUserById(id)
        return userRepository.save(user)
    }

    fun deleteUser(id: Long) {
        val user = getUserById(id)
        userRepository.delete(user)
    }
}
